File: src/test/java/com/dg/watcher/history/HistoryTest.java
package com.dg.watcher.history;

import hudson.model.AbstractProject;
import org.junit.Test;
import static org.hamcrest.Matchers.equalTo;
import static org.hamcrest.Matchers.is;
import static org.junit.Assert.assertThat;
import static org.mockito.Mockito.mock;


public class HistoryTest {
    @Test
    public void should_specify_the_shown_icon() {
        assertThat(history().getIconFileName(), is(equalTo("graph.gif")));
    }

    @Test
    public void should_specify_the_shown_label() {
        assertThat(history().getDisplayName(), is(equalTo("Apk History")));
    }

    @Test
    public void should_specify_the_used_domain() {
        assertThat(history().getUrlName(), is(equalTo("apk_history")));
    }

    private History history() {
        return new History(mock(AbstractProject.class));
    }
}

File: src/test/java/com/dg/watcher/history/HistoryTest.kt
package com.dg.watcher.history

import com.nhaarman.mockito_kotlin.mock
import org.hamcrest.Matchers.`is`
import org.hamcrest.Matchers.equalTo
import org.junit.Assert.assertThat
import org.junit.Test


class HistoryTest {
    @Test
    fun `Should specify the shown icon`() =
            assertThat(history().iconFileName, `is`(equalTo("graph.gif")))

    @Test
    fun `Should specify the shown label`() =
            assertThat(history().displayName, `is`(equalTo("Apk History")))

    @Test
    fun `Should specify the used domain`() =
            assertThat(history().urlName, `is`(equalTo("apk_history")))

    private fun history() = History(mock())
}